### ISSUE TITLE
Crash in analysisd when handling openldap (and maybe others) logs

### ISSUE THREAD

**Issue by mcm** [wazuh/wazuh#2453]
## Bug report

**OS**
Ubuntu 16.04

**Wazuh version**
3.8.0

**Wazuh API version**
n/a

**Install type**
manager

**Install method**
packages

**Log sample**

```
Jan 24 17:00:05 ldap slapd[1770]: conn=81365 fd=62 ACCEPT from IP=10.xxx.xxx.xxx:56156 (IP=0.0.0.0:636)
Jan 24 17:00:05 ldap slapd[1770]: conn=81365 fd=62 TLS established tls_ssf=256 ssf=256
Jan 24 17:00:05 ldap slapd[1770]: conn=81365 op=0 BIND dn="cn=xxx,ou=xxx,dc=xxx,dc=xxx" method=128
Jan 24 17:00:05 ldap slapd[1770]: conn=81365 op=0 BIND dn="cn=xxx,ou=xxx,dc=xxx,dc=xxx" mech=SIMPLE ssf=0
Jan 24 17:00:05 ldap slapd[1770]: conn=81365 op=0 RESULT tag=97 err=0 text=
Jan 24 17:00:05 ldap slapd[1770]: conn=81365 op=1 SRCH base="dc=xxx,dc=xxx" scope=2 deref=3 filter="(&(!(ou=xxx))(|(homePhone=*)(mobile=*)(telephoneNumber=*)))"
Jan 24 17:00:05 ldap slapd[1770]: conn=81365 op=1 SRCH attr=cn homePhone mail mobile ou telephoneNumber
Jan 24 17:00:05 ldap slapd[1770]: conn=81365 op=1 SEARCH RESULT tag=101 err=0 nentries=748 text=
```

Bug description
----------------

As best as I can tell, something is causing `FreeACMStore` more than once, which causes a ossec-analysisd to crash:

```
2019/01/24 17:00:05 ossec-analysisd[5610] accumulator.c:181 at Accumulate(): DEBUG: accumulator: DEBUG: (ldap openldap 81365) updated stored_data->dstip to 0.0.0.0
2019/01/24 17:00:05 ossec-analysisd[5610] accumulator.c:185 at Accumulate(): DEBUG: accumulator: DEBUG: (ldap openldap 81365) updated stored_data->srcip to 10.xxx.xxx.xxx
2019/01/24 17:00:05 ossec-analysisd[5610] accumulator.c:214 at Accumulate(): DEBUG: accumulator: DEBUG: Added stored data for ldap openldap 81365
2019/01/24 17:00:05 ossec-analysisd[5610] state.c:96 at w_analysisd_write_state(): DEBUG: Updating state file.
2019/01/24 17:00:05 ossec-analysisd[5610] accumulator.c:94 at Accumulate(): DEBUG: accumulator: DEBUG: No id available
2019/01/24 17:00:05 ossec-analysisd[5610] accumulator.c:125 at Accumulate(): DEBUG: accumulator: DEBUG: Lookup for 'ldap openldap 81365' found a stored value!
2019/01/24 17:00:05 ossec-analysisd[5610] accumulator.c:147 at Accumulate(): DEBUG: accumulator: DEBUG: (ldap openldap 81365) updated lf->dstip to 0.0.0.0
2019/01/24 17:00:05 ossec-analysisd[5610] accumulator.c:151 at Accumulate(): DEBUG: accumulator: DEBUG: (ldap openldap 81365) updated lf->srcip to 10.xxx.xxx.xxx
2019/01/24 17:00:05 ossec-analysisd[5610] accumulator.c:173 at Accumulate(): DEBUG: accumulator: DEBUG: (ldap openldap 81365) updated stored_data->dstuser to portal
2019/01/24 17:00:05 ossec-analysisd[5610] accumulator.c:300 at FreeACMStore(): DEBUG: accumulator: DEBUG: Freeing an accumulator struct.
2019/01/24 17:00:05 ossec-analysisd[5610] accumulator.c:207 at Accumulate(): DEBUG: accumulator: DEBUG: Updated stored data for ldap openldap 81365
2019/01/24 17:00:05 ossec-analysisd[5610] accumulator.c:125 at Accumulate(): DEBUG: accumulator: DEBUG: Lookup for 'ldap openldap 81365' found a stored value!
2019/01/24 17:00:05 ossec-analysisd[5610] accumulator.c:147 at Accumulate(): DEBUG: accumulator: DEBUG: (ldap openldap 81365) updated lf->dstip to �
2019/01/24 17:00:05 ossec-analysisd[5610] accumulator.c:151 at Accumulate(): DEBUG: accumulator: DEBUG: (ldap openldap 81365) updated lf->srcip to �
2019/01/24 17:00:05 ossec-analysisd[5610] accumulator.c:300 at FreeACMStore(): DEBUG: accumulator: DEBUG: Freeing an accumulator struct.
*** Error in `/var/ossec/bin/ossec-analysisd': free(): invalid pointer: 0x00007f83100000b8 ***
Segmentation fault (core dumped)
```

This wasn't happening in 3.7.2 and my GUT tells me that it's related to ae29d7ecf3cd15fcd45a0ca5bea3ad67d06fcd79 though I haven't been able to confirm that for sure. Running `ossec-logtest` on that log sample will segfault it, though I think in a slightly different way maybe. In both cases, it seems to be crashing immediately after the second BIND log. Removing the second bind from the log sample, it then crashes on the RESULT line in the same manner. 

Please let me know if I can provide any additional information to help with debugging.

**Comment by snaow** [comment]
Hi @mcm,

Thanks for the report, I can confirm it is happening as well on my environment, pretty easy to recreate.

The team is already working on fixing it, I will update you as soon as I can with any progress.

**Comment by mcm** [comment]
Awesome, thanks for the quick response. I'm happy to test when you think you've got it fixed if necessary.

**Comment by vikman90** [comment]
Hi @mcm,

We are very grateful to you for reporting this error and your detailed description, that let us find the bug quickly.

We proposed a PR to fix the bug: #2456. We used the input logs that you send us above and I confirm that Analysisd no longer crashes with this fix. It would be great if you could also try it. Hope that this definitely solves the problem.

Thanks for your help, we really appreciate it.

**Comment by mcm** [comment]
Thanks again for the quick turnaround on this. I can confirm that this patch fixed the issue for us.

Just so I understand, should I expect an eventual 3.8.2 with this fix, or should I expect it in 3.9?

Regardless, appreciate everything and thanks!

**Comment by vikman90** [comment]
@mcm Thank you for testing our fix. We have merged it into branch 3.8 for a next version 3.8.2. 

Since this is a bug that is impacting the manager, we are going to release 3.8.2 by mid next week.

Let me close this issue. Please don't hesitate to reopen it if you have any further problem or question.

Thank you again.
Best regards.
